src/common/hobject_fmt.h

```diff
@@ -39,14 +47,13 @@ template <> struct fmt::formatter<hobject_t> {
       return fmt::format_to(ctx.out(), "MAX");
     }
 
-    std::string v;
-    append_out_escaped(ho.nspace, &v);
-    v.push_back(':');
-    append_out_escaped(ho.get_key(), &v);
-    v.push_back(':');
-    append_out_escaped(ho.oid.name, &v);
-
-    return fmt::format_to(ctx.out(), "{}:{:08x}:{}:{}", static_cast<uint64_t>(ho.pool),
-			  ho.get_bitwise_key_u32(), v, ho.snap);
+    fmt::format_to(
+	ctx.out(), FMT_COMPILE("{}:{:08x}:"), static_cast<uint64_t>(ho.pool),
+	ho.get_bitwise_key_u32());
+    append_sanitized(ctx, ho.nspace, ':');
+    append_sanitized(ctx, ho.get_key(), ':');
+    append_sanitized(ctx, ho.oid.name);
+    return fmt::format_to(ctx.out(), FMT_COMPILE(":{}"), ho.snap);
   }
 };
+}  // namespace fmt
```

Comment:
Are there unit tests to guarantee that this doesn't change the resulting string?

Reply:
There already is some external test code that I used during development (linked in commit messages).
I will add a unit test executable to this PR, as well.